runtime/tests/checker/entitlements_test.go

```diff
@@ -1975,6 +1975,21 @@ func TestCheckEntitlementInheritance(t *testing.T) {
 		assert.NoError(t, err)
 	})
 
+	t.Run("valid interface", func(t *testing.T) {
+		t.Parallel()
+		_, err := ParseAndCheck(t, `
+			entitlement E
+			struct interface I {
+				access(E) fun foo() 
+			}
+			struct interface S: I {
+				access(E) fun foo() 
+			}
+		`)
+
+		assert.NoError(t, err)
+	})
+
```

Comment:
I remember we once discussed about entitlements coming from default functions: I think it was something like taking an `auth{M}` reference to a composite `T` where `T` conforms to interface `I`, and `I` have default function(s) with entitlement mapping `M`, while `T` doesn't have any entitlements directly specified on it. 

Maybe also add a test for the same, but this time for entitlements coming from inherited default functions. Think it should also work out of the box.